### ISSUE TITLE
App crashes on every YT movie I try to download.

### ISSUE THREAD

**Issue by turboteckel** [TeamNewPipe/NewPipe#1325]
As far as I'm aware I haven't changed anything since yesterday when it was all working fine. Just uninstalled and reinstalled from fdroid and the problem is still there. 

Here's the error report. Any idea what causes this?

```
{
   "user_action": "ui error",
   "request": "App crash, UI failure",
   "content_language": "GB",
   "service": "none",
   "package": "org.schabi.newpipe",
   "version": "0.13.2",
   "os": "Linux Android 7.0 - 24",
   "time": "2018-04-17 10:43",
   "exceptions": [
      "java.lang.RuntimeException: javax.net.ssl.SSLHandshakeException: Connection closed by peer\n\tat us.shandian.giga.get.DownloadManagerImpl$Initializer.run(DownloadManagerImpl.java:350)\nCaused by: javax.net.ssl.SSLHandshakeException: Connection closed by peer\n\tat com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)\n\tat com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:364)\n\tat com.android.okhttp.Connection.connectTls(Connection.java:1510)\n\tat com.android.okhttp.Connection.connectSocket(Connection.java:1458)\n\tat com.android.okhttp.Connection.connect(Connection.java:1413)\n\tat com.android.okhttp.Connection.connectAndSetOwner(Connection.java:1707)\n\tat com.android.okhttp.OkHttpClient$1.connectAndSetOwner(OkHttpClient.java:133)\n\tat com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:466)\n\tat com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:371)\n\tat com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:503)\n\tat com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:438)\n\tat com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponseCode(HttpURLConnectionImpl.java:567)\n\tat com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getResponseCode(DelegatingHttpsURLConnection.java:105)\n\tat com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java)\n\tat us.shandian.giga.get.DownloadManagerImpl$Initializer.run(DownloadManagerImpl.java:313)\n\tSuppressed: javax.net.ssl.SSLHandshakeException: Connection closed by peer\n\t\t... 15 more\n\t\tSuppressed: javax.net.ssl.SSLHandshakeException: SSL handshake aborted: ssl=0x7e425ffc00: I\/O error during system call, Connection reset by peer\n\t\t\t... 15 more\njavax.net.ssl.SSLHandshakeException: Connection closed by peer\n\tat com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)\n\tat com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:364)\n\tat com.android.okhttp.Connection.connectTls(Connection.java:1510)\n\tat com.android.okhttp.Connection.connectSocket(Connection.java:1458)\n\tat com.android.okhttp.Connection.connect(Connection.java:1413)\n\tat com.android.okhttp.Connection.connectAndSetOwner(Connection.java:1707)\n\tat com.android.okhttp.OkHttpClient$1.connectAndSetOwner(OkHttpClient.java:133)\n\tat com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:466)\n\tat com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:371)\n\tat com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:503)\n\tat com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:438)\n\tat com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponseCode(HttpURLConnectionImpl.java:567)\n\tat com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getResponseCode(DelegatingHttpsURLConnection.java:105)\n\tat com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java)\n\tat us.shandian.giga.get.DownloadManagerImpl$Initializer.run(DownloadManagerImpl.java:313)\n\tSuppressed: javax.net.ssl.SSLHandshakeException: Connection closed by peer\n\t\t... 15 more\n\t\tSuppressed: javax.net.ssl.SSLHandshakeException: SSL handshake aborted: ssl=0x7e425ffc00: I\/O error during system call, Connection reset by peer\n\t\t\t... 15 more\n"
   ],
   "user_comment": ""
}

```

**Comment by theScrabi** [comment]
Looks like a certificate error. Have you checked the youtube certificate with your browser?

**Comment by turboteckel** [comment]
Certificate looks fine as far as I can tell.

**Comment by theScrabi** [comment]
@theassassin what could be the reason for the handhake to fail? Fals authority maybe cert pining? We don't fo that tho. Or some intercepting by a provider?

**Comment by TheAssassin** [comment]
@theScrabi to me, it looks like the connection was aborted during the handshake. This might have different reasons. For example, the server may have terminated while the handshake wasn't finished, but the error description doesn't contain any specific information. All I can tell is that it was not the client who aborted the connection.

Do we have an idea what URL it has tried to download? Are there any "anti-virus" gateways or proxies interfering with the request?

**Comment by TobiGr** [comment]
this should be fixed by #1759 . Please let us know if you still get this error with NewPipe 0.15.0 (not yet released).
